File: PyProject/sparkproject/dataframe_sql/df.py
# coding: utf-8


from pyspark.sql import SparkSession

if __name__ == '__main__':
    # conf = SparkConf().setAppName("sql").setMaster("local[*]")
    spark = SparkSession.builder.getOrCreate()
    df = spark.read.csv("./gdp.csv")

    df.registerTempTable("gdp")

    df2 = spark.sql("select * from gdp limit 1000")
    # filter
    # df3 = df2.filter("_c2='1969'")
    # df3 = df2.select("_c0").distinct()

    df3 = df2.select("_c0").groupBy("_c3").count()
    df3.show()
    # drop table
    spark.catalog.dropTempView("gdp")
    spark.stop()

File: src/main/scala/org/bigdata/df/DF.scala
package org.bigdata.df

import org.apache.spark.sql.{DataFrame, SparkSession}
import org.apache.spark.SparkConf

object DF {
  def main(args: Array[String]): Unit = {
    val conf: SparkConf = new SparkConf().setAppName("sql").setMaster("local[*]")
    val spark: SparkSession = SparkSession.builder().appName("sql").config(conf).getOrCreate()

    // read files
    val df: DataFrame = spark.read.csv("***")
    val df1: DataFrame = spark.read.parquet("***")
    val df2: DataFrame = spark.read.orc("***")
    val df3: DataFrame = spark.read.json("***")
    val df4: DataFrame = spark.read.text("***")


    val df5: DataFrame = spark.read
      .format("jdbc")
      .option("url", "")
      .option("dbtable", "")
      .option("user", "")
      .option("password", "")
      .load()


  }
}
